### PR TITLE
Match n64dd text in code

### PR DESCRIPTION
**Pull request by Dragorn421**
I started this without knowing where I'd end up and now I know what comes next is disassembling the n64dd segment (`func_800ADA80` gives its vram start, vrom range and bss vram range), and matching other functions using n64dd stuff like Main, game.c, ...

I didn't expect this rabbit hole so I'm stopping here for now, just with matched n64dd text in code

### CODE REVIEW

**Comment by cadmic** [comment]
I don't personally plan on setting up disassembly for n64dd, because that's a lot of work and it's already fully matching at https://github.com/Decompollaborate/n64dd. So I'm hoping we could slowly import that code from there

**Comment by cadmic** [comment]
on second thought, generating the disassembly config could probably be automated from that repo, so maybe that is the way to go after all?

File: src/code/code_n64dd_800AD410.c [hunk 1]
```diff
@@ -1 +1,19 @@
+#include "ultra64.h"
+#include "z64dma.h"
+#include "n64dd.h"
 
+void func_800ADA80(void) {
+    if (!B_80121AE0) {
+        DmaMgr_RequestSync(func_801C7740_unknown, (uintptr_t)D_00B8ADA0, D_00B9DA70 - D_00B8ADA0);
```
Comment (cadmic) [inline src/code/code_n64dd_800AD410.c:7]: This is _n64dd_SegmentStart/_n64dd_SegmentRomStart. Maybe we could include the n64dd baserom .o file in the spec for now?

File: src/code/code_n64dd_800AD4C0.c [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+        if (B_80121AF0->unk_04 != NULL) {
+            B_80121AF0->unk_04();
+        }
+    }
+    B_80121AF0 = NULL;
+}
+
+n64ddStruct_800FF4B0_pointers* func_800ADBD0(void) {
+    D_800FF4B0.unk_04 = 0;
+    D_800FF4B0.unk_08 = (RegEditor*)gRegEditor;
```
Comment (cadmic) [inline src/code/code_n64dd_800AD4C0.c:27]: why the cast?

**Review by cadmic — COMMENTED** [review]
build seems to be failing for non-trivial reasons

File: src/code/code_n64dd_800AD4C0.c (hunk 2 above)
Comment (Dragorn421) [reply src/code/code_n64dd_800AD4C0.c:27]: *points at m2c and surrounding mess*
will remove

File: undefined_syms.txt [hunk 3]
```diff
@@ -9,3 +9,15 @@ D_0F000000 = 0x0F000000;
...
+#if PLATFORM_N64
+func_801C7740_unknown = 0x801C7740;
+B_80121AE0 = 0x80121AE0;
+B_80121AE1 = 0x80121AE1;
+D_00B8ADA0 = 0x00B8ADA0;
+D_00B9DA70 = 0x00B9DA70;
+D_801DA410 = 0x801DA410;
+D_801E8090 = 0x801E8090;
+B_80121AE2 = 0x80121AE2;
+D_800FF4B0 = 0x800FF4B0;
```
Comment (cadmic) [inline undefined_syms.txt:22]: (Sorry, missed this earlier)
Some of these don't need to be undefined symbols:

0x800FF4B0: in code_n64dd_800AD4C0 .data
0x80121AE0: in src/code/code_n64dd_800AD410 .bss
0x80121AE1: in src/code/code_n64dd_800AD410 .bss
0x80121AE2: in src/code/code_n64dd_800AD410 .bss

Of course these file splits were invented by me so not sure which code_n64dd file they should go in. But the point is I think we should define them now

File: src/code/code_n64dd_800AD4C0.c [hunk 4]
```diff
@@ -1,3 +1,90 @@
+#include "global.h"
 #include "n64dd.h"
 
+// TODO functions of unknown prototype
+extern char func_801C8510_unknown[];
+extern char func_800AE170_unknown[];
+extern char func_800ADCD8_unknown[];
```
Comment (cadmic) [inline src/code/code_n64dd_800AD4C0.c:7]: Seems like 800AE170 is `Fault_DrawText` and `800ADCD8` is `Fault_WaitForInput` from your other PR. Not sure how you want to handle it but it seems fine for now

**Review by cadmic — APPROVED** [review]
nice cleanups in the last set of changes

File: src/code/code_n64dd_800AD4C0.c (hunk 4 above)
Comment (Dragorn421) [reply src/code/code_n64dd_800AD4C0.c:7]: yeah I noticed, I want to handle it by waiting for either PR to be merged and then updating the other

**Review by fig02 — APPROVED** [review]

**Pull request merged**